internal/mutator/pod/agent/injector.go

```diff
@@ -364,9 +364,27 @@ func (i *injector) shouldInjectContainer(ctx context.Context, pod *corev1.Pod, n
 		return false, fmt.Errorf("getting Namespace %q for policy matching: %w", namespace, err)
 	}
 
+	// Checking if there is any overlapping with the volumes we want to mount and the volumes already present.
+	duplicate := isVolumeNameDuplicatePresent(containerToInject.VolumeMounts, pod.Spec.Volumes)
```

Comment:
I think we should be comparing `pod.Spec.Volumes` with output of this code:
```go
  volumes := []corev1.Volume{}
  for _, v := range i.container.VolumeMounts {
    // TODO We should check if the volume is already present
    volumes = append(volumes, corev1.Volume{
      Name: v.Name,
      VolumeSource: corev1.VolumeSource{
        EmptyDir: &corev1.EmptyDirVolumeSource{},
      },
    })
  }
```

I'd move this to a function and then compare it.

Do something like: `toEmptyDirVolumes(standardVolumeMounts())`.

Then when mutating, you can do:
```go
pod.Spec.Volumes = append(pod.Spec.Volumes, toEmptyDirVolumes(containerToInject.VolumeMounts))...)
```

Reply:
I change it and it improves a lot the code!